doc/docgen.rst

```diff
@@ -229,6 +229,146 @@ Output::
 Note that the `jsondoc`:option: command outputs its JSON without pretty-printing it,
 while `jsondoc0`:option: outputs pretty-printed JSON.
 
+
+Referencing Nim symbols: simple documentation links
+===================================================
+
+You can reference Nim identifiers from Nim documentation comments (currently
+only inside one ``.nim`` file).
+This pertains to any exported symbol like `proc`, `const`, `iterator`, etc.
+Syntax for referencing is basically a normal RST one: addition of
+underscore `_` to a *link text*.
+Link text is either one word or a group of words enclosed by backticks `\``.
+Link text will be displayed *as is* while *link target* will be set to
+the anchor of Nim symbol that corresponds to link text (anchors' format
+is described in `HTML anchor generation`_ section).
+
+If you have a constant:
+
+.. code:: Nim
+
+   const pi* = 3.14
+
+then it should be referenced in one of the 2 forms:
+
+A. non-qualified (no symbol kind specification)::
+     pi_
+B. qualified (with symbol kind specification)::
+     `const pi`_
+
+For routine kinds there are more options. Consider this definition:
+
+.. code:: Nim
+
+   proc foo*(a: int, b: float): string
+
+Generally following syntax is allowed for referencing `foo`:
+
+*  short (without parameters):
+
+   A. non-qualified::
+
+        foo_
```

Comment:
maybe mention this is the preferred form because:
* otherwise ppl will get confused as to which form to use by default
* simpler = better
* more robust to API changes (eg proc => template, adding optional params etc)

Reply:
i think one suggestion is enough. Besides some people may like to provide parameters to procs — I would prefer that in most cases also.